codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ValueEntitySourceGenerator.scala

```diff
@@ -0,0 +1,203 @@
+        line <>
+        method(
+          "public",
+          qualifiedType(entity.state.fqn),
+          "emptyState",
+          Nil,
+          emptyDoc
+        )(
+          "return" <+> qualifiedType(entity.state.fqn) <> ".getDefaultInstance();"
+        ) <>
+        line <>
+        line <>
+        ssep(
+          service.commands.toSeq.map { command =>
+            "@Override" <>
+            line <>
+            method(
+              "public",
+              "Effect" <> angles(qualifiedType(command.outputType)),
+              lowerFirst(command.fqn.name),
+              List(
+                qualifiedType(entity.state.fqn) <+> "currentState",
+                qualifiedType(command.inputType) <+> "command"
+              ),
+              emptyDoc
+            ) {
+              "return effects().error" <> parens(
+                "\"The command handler for `" + command.fqn.name + "` is not implemented, yet\""
+              ) <> semi
+            }
+          },
+          line <> line
+        )
+      }
+    )
+  }
+
+  private[codegen] def valueEntityHandler(service: ModelBuilder.EntityService,
+                                          entity: ModelBuilder.ValueEntity,
+                                          packageName: String,
+                                          className: String): Document = {
+
+    val imports = generateImports(
+      service.commands,
+      Some(entity.state),
+      packageName,
+      otherImports = Seq(
+        "com.akkaserverless.javasdk.impl.AnySupport",
+        "com.akkaserverless.javasdk.impl.EntityExceptions",
+        "com.akkaserverless.javasdk.impl.FailInvoked$",
+        "com.akkaserverless.javasdk.lowlevel.ValueEntityHandler",
+        "com.akkaserverless.javasdk.valueentity.CommandContext",
+        "com.akkaserverless.javasdk.valueentity.ValueEntityBase",
+        "scalapb.UnknownFieldSet"
+      )
+    )
+
+    pretty(
+      managedCodeCommentString <> line <> // fixme: why this can't be in string interpolation?
+      s"""|package $packageName;
+          |
+          |$imports
+          |
+          |/** A value entity handler */
+          |public class ${className}Handler extends ValueEntityHandler {
+          |
+          |  final $className entity;
+          |  
+          |  ${className}Handler() {
+          |    this.entity = new $className();
+          |  }
+          |
+          |
+          |  @Override
+          |  public ValueEntityBase.Effect<? extends GeneratedMessageV3> handleCommand(
+          |      Any command, Any state, CommandContext<Any> context) throws Throwable {
+          |    try {
+          |      return invoke(command, state, context);
+          |    } catch (Exception e) {
+          |      if (e.getClass().isAssignableFrom(FailInvoked$$.class)) {
+          |        throw e;
+          |      } else {
+          |        throw e.getCause();
+          |      }
+          |    }
+          |  }
+          |}""".stripMargin
```

Comment:
BTW, this is what a mean by writing a codegen where we can recognize the generated code directly instead of continually have to parse `<>`, `<+>`, `line` and all that syntax.